kyaml/fn/framework/patch.go

```diff
@@ -180,7 +182,16 @@ func (cpt ContainerPatchTemplate) apply(matches []*yaml.RNode) error {
 				return nil
 			}
 			for j := range patches {
-				_, err := merge2.Merge(patches[j], node, yaml.MergeOptions{})
+				merger := walk.Walker{
+					Sources:      []*yaml.RNode{node, patches[j]}, // dest, src
+					Visitor:      merge2.Merger{},
+					MergeOptions: yaml.MergeOptions{},
+					Schema: openapi.SchemaForResourceType(yaml.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Pod",
```

Comment:
This is a bit surprising. Why is this only on Pod?
I know it's rare, but could this also be required to be done on PodList, PodTemplate, or PodTemplateList?

Reply:
What we're doing here is grabbing the OpenAPI schema for `Container`, because `openapi.SchemaForResourceType` doesn't expose fields--just top-level resources (with some exceptions because of the heuristic it is using to determine that). We could grab the schema off any one of the types you mention; the result would be the same.